packages/shorebird_cli/lib/src/commands/patch/patch_command.dart

```diff
@@ -285,14 +290,33 @@ NOTE: this is ${styleBold.wrap('not')} recommended. Asset changes cannot be incl
 
     File? patchArtifactFile;
     final Release release;
-    if (results.wasParsed('release-version')) {
+    final releasePlatform = patcher.releaseType.releasePlatform;
+    if (useLatestRelease) {
+      final releases = await codePushClientWrapper.getReleases(appId: appId);
+      releases
+        ..removeWhere(
+          (release) => !release.platformStatuses.keys.contains(releasePlatform),
+        )
+        ..sort(
+          (a, b) => Version.parse(b.version).compareTo(
+            Version.parse(a.version),
+          ),
+        );
```

Comment:
I think this might be worth pulling out into an extension and unit testing separately. Looking at the pub_semver package, it's not clear how this would handle build numbers or versions that aren't technically semver (e.g., `1.0`)

Reply:
Yup good call, updated to include dedicated unit tests for sort order.